tlsr9/CMakeLists.txt

```diff
@@ -70,50 +89,88 @@ if(CONFIG_BT_B9X OR CONFIG_IEEE802154_TELINK_B9X)
 		common/b9x_rf_power.c
 	)
 endif()
+if(CONFIG_BT_TLX OR CONFIG_IEEE802154_TELINK_TLX)
+	zephyr_library_sources(
+		drivers/${SOC}/aes.c
+		common/b9x_rf_power.c
+	)
+endif()
 
 # ADC driver reference sources
 zephyr_library_sources_ifdef(CONFIG_ADC_TELINK_B9X drivers/${SOC}/adc.c)
 zephyr_library_sources_ifdef(CONFIG_ADC_TELINK_B9X drivers/${SOC}/gpio.c)
+zephyr_library_sources_ifdef(CONFIG_ADC_TELINK_TLX drivers/${SOC}/adc.c)
+zephyr_library_sources_ifdef(CONFIG_ADC_TELINK_TLX drivers/${SOC}/gpio.c)
 
 # USB driver reference sources
 zephyr_library_sources_ifdef(CONFIG_USB_TELINK_B9X drivers/${SOC}/gpio.c)
 zephyr_library_sources_ifdef(CONFIG_USB_TELINK_B9X drivers/${SOC}/usbhw.c)
+zephyr_library_sources_ifdef(CONFIG_USB_TELINK_TLX drivers/${SOC}/gpio.c)
+zephyr_library_sources_ifdef(CONFIG_USB_TELINK_TLX drivers/${SOC}/usbhw.c)
 
 #PM driver dependency
+if (NOT CONFIG_SOC_RISCV_TELINK_TL321X)
 zephyr_library_sources_ifdef(CONFIG_PM drivers/${SOC}/stimer.c)
+endif()
 zephyr_library_sources_ifdef(CONFIG_PM common/sleep.c)
 
 #BLE flash reference sources
-if (CONFIG_IEEE802154_TELINK_B9X)
+if (CONFIG_IEEE802154_TELINK_B9X OR CONFIG_IEEE802154_TELINK_TLX)
 
-zephyr_include_directories(
+zephyr_include_directories_ifdef(CONFIG_IEEE802154_TELINK_B9X
 	ble
-	ble/vendor/controller
+	ble/vendor/controller/b9x
+)
+
+zephyr_include_directories_ifdef(CONFIG_IEEE802154_TELINK_TLX
+	ble
+	ble/vendor/controller/tlx
 )
 
 endif()
 
 # BLE stack reference sources
-if (CONFIG_BT_B9X)
+if (CONFIG_BT_B9X OR CONFIG_BT_TLX)
+
+zephyr_include_directories_ifdef(CONFIG_BT_B9X
+	ble
+	ble/vendor/controller/b9x
+)
 
-zephyr_include_directories(
+zephyr_include_directories_ifdef(CONFIG_BT_TLX
 	ble
-	ble/vendor/controller
+	ble/vendor/controller/tlx
 )
 
-zephyr_library_sources(
+zephyr_library_sources_ifdef(CONFIG_BT_B9X
 	ble/common/utility.c
-	ble/vendor/controller/b9x_bt.c
-	ble/vendor/controller/b9x_bt_init.c
-	ble/vendor/controller/b9x_bt_buffer.c
-	ble/vendor/controller/b9x_bt_flash.c
+	ble/vendor/controller/b9x/b9x_bt.c
+	ble/vendor/controller/b9x/b9x_bt_init.c
+	ble/vendor/controller/b9x/b9x_bt_buffer.c
+	ble/vendor/controller/b9x/b9x_bt_flash.c
 )
 
-endif() # CONFIG_BT_B9X
+zephyr_library_sources_ifdef(CONFIG_BT_TLX
+	ble/common/utility.c
+	ble/vendor/controller/tlx/tlx_bt.c
+	ble/vendor/controller/tlx/tlx_bt_init.c
+	ble/vendor/controller/tlx/tlx_bt_buffer.c
+	ble/vendor/controller/tlx/tlx_bt_flash.c
+)
+
+endif() # (CONFIG_BT_B9X OR CONFIG_BT_TLX)
+
+add_definitions(-DSTD_GCC)
```

Comment:
Yes. This macro is provided by the Driver SDK team. It must be provided when using non-Andes GCC.

Reply:
Since `-DSTD_GCC` is only valid for TL321X, we may add conditional lines like this
`
if (CONFIG_SOC_RISCV_TELINK_TL321X)
add_definitions(-DSTD_GCC)
endif()`